src/mem.rs

```diff
@@ -419,6 +453,17 @@ impl Decompress {
         }
     }
 
+    /// Specifies the decompression dictionary to use.
+    #[cfg(feature = "zlib")]
+    pub fn set_dictionary(&mut self, dictionary: &[u8]) {
+        let stream = &mut *self.inner.stream_wrapper;
+        let rc = unsafe {
+            ffi::inflateSetDictionary(stream, dictionary.as_ptr(), dictionary.len() as ffi::uInt)
+        };
+
+        assert_eq!(rc, ffi::MZ_OK);
```

Comment:
I was thinking it would be better to return `Result<u32, DecompressError>` from this function so that error in case of incorrect dictionary or in case of calling `set_dictionary` at incorrect time (it has to be called only immediately after `Z_NEED_DICT`) could be still handled gracefully.

Reply:
I believe I originally implemented these methods before flate2 returned `Result` and simply had asserts against MZ_OK.

Returning an actual `Result` is definitely better, I assume it's safe to presume the adler code will already be populated from a previous call to inflate. So if `inflateSetDictionary` works without error I can return that.